### PR TITLE
[rllib] Updating Repeated space. Allowing numpy arrays and adding representation

### PR DESCRIPTION
**Pull request by brunnelu**
## Why are these changes needed?

The Repeated space does not allow numpy arrays. This is inefficient for many environments, because a numpy array would have to be converted to a list (in env) and again by rllibs sample collection. 

The representation is added in case someone wants to use this space in a print statement.

## Related issue number

None, optimizations only

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by Sertingolix** [comment]
Added usability of new repeated space in model predecessor and run tests.

File: rllib/utils/spaces/repeated.py [hunk 1]
```diff
@@ -27,5 +28,9 @@ def sample(self):
         ]
 
     def contains(self, x):
-        return (isinstance(x, list) and len(x) <= self.max_len
+        return (isinstance(x, (list,np.ndarray))
```
Comment (sven1977) [inline rllib/utils/spaces/repeated.py:31]:
```suggestion
        return (isinstance(x, (list, np.ndarray))
```

**Review by sven1977 — APPROVED** [review]
Looks good @Sertingolix , thanks for the fix! :)

File: rllib/utils/spaces/repeated.py [hunk 2]
```diff
@@ -27,5 +28,9 @@ def sample(self):
...
+                and len(x) <= self.max_len
```
Comment (sven1977) [inline rllib/utils/spaces/repeated.py:32]:
```suggestion
        return (isinstance(x, (list, np.ndarray)) and len(x) <= self.max_len
```

File: rllib/models/preprocessors.py [hunk 3]
```diff
@@ -316,7 +316,7 @@ def transform(self, observation: TensorType) -> np.ndarray:
     @override(Preprocessor)
     def write(self, observation: TensorType, array: np.ndarray,
               offset: int) -> None:
-        if not isinstance(observation, list):
+        if not isinstance(observation, (list,np.ndarray)):
```
Comment (sven1977) [inline rllib/models/preprocessors.py:319]:
```suggestion
        if not isinstance(observation, (list, np.ndarray)):
```

File: rllib/policy/torch_policy.py [hunk 4]
```diff
@@ -1067,9 +1067,10 @@ def _worker(shard_idx, model, sample_batch, device):
                 with lock:
                     results[shard_idx] = (all_grads, grad_info)
             except Exception as e:
+                import traceback
                 with lock:
                     results[shard_idx] = (ValueError(
-                        e.args[0] + "\n" +
+                        e.args[0] + "\n traceback" + traceback.format_exc() + "\n" +
```
Comment (sven1977) [inline rllib/policy/torch_policy.py:1073]:
```suggestion
                        e.args[0] + "\n traceback" + traceback.format_exc() +
                        "\n" +
```

**Comment by sven1977** [comment]
Waiting for LINT fixes to pass. ...

For the future, to avoid the LINTer CI tests to break, you can run `ci/travis/format.sh` in your ray git repo locally before committing/pushing and it should give you those files+linenumbers that need lint-fixing.

**Comment by Sertingolix** [comment]
Sure. Will do so next time ;)

**Pull request merged**